docs/resources/environment.md

```diff
@@ -0,0 +1,39 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "axual_environment Resource - terraform-provider-axual"
+subcategory: ""
+description: |-
+  Environments are used typically to support the application lifecycle, as it is moving from Development to Production.  In Self Service, they also allow you to test a feature in isolation, by making the environment Private. Read more: https://docs.axual.io/axual/2023.1/self-service/environment-management.html#managing-environments
+---
+
+# axual_environment (Resource)
+
+Environments are used typically to support the application lifecycle, as it is moving from Development to Production.  In Self Service, they also allow you to test a feature in isolation, by making the environment Private. Read more: https://docs.axual.io/axual/2023.1/self-service/environment-management.html#managing-environments
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `authorization_issuer` (String) This indicates if any deployments on this environment should be AUTO approved or requires approval from Stream Owner. For private environments, only AUTO can be selected.
+- `color` (String) The color used display the environment
+- `instance` (String) The id of the instance where this environment should be deployed.
+- `name` (String) A suitable name identifying this environment. This must be in the format string-string (Alphabetical characters, digits and the following characters are allowed: `- `,` _` ,` .`)
+- `owners` (String) The id of the team owning this environment.
+- `short_name` (String) A short name that will uniquely identify this environment. The short name should be between 3 and 20 characters. no special characters are allowed.
+- `visibility` (String) Thi Private environments are only visible to the owning group (your team). They are not included in dashboard visualisations.
+
+### Optional
+
+- `description` (String) A text describing the purpose of the environment.
+- `partitions` (Number) Defines the number of partitions configured for every stream of this tenant. This is an optional field. If not specified, default value is 12
+- `properties` (Map of String) Environment-wide properties for all topics and applications.
+- `retention_time` (Number) The time in milliseconds after which the messages can be deleted from all streams. This is an optional field. If not specified, default value is 7 days (604800000).
+
+### Read-Only
+
+- `id` (String) Environment unique identifier
+
+
```

Comment:
Other resources have "Example usage" and "Import" sections. Can we add them for environment please?

https://registry.terraform.io/providers/Axual/axual/latest/docs/resources/stream